examples/http_body.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proxy_wasm::traits::*;
+use proxy_wasm::types::*;
+
+#[no_mangle]
+pub fn _start() {
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_http_context(|_, _| -> Box<dyn HttpContext> { Box::new(HttpBody {}) });
+}
+
+struct HttpBody {}
+
+impl Context for HttpBody {}
+
+impl HttpContext for HttpBody {
+    fn on_http_response_headers(&mut self, _: usize) -> Action {
+        // If there is a content-length header and we change the length of
+        // the body later, then clients will break. So remove it.
+        // We must do this here, because once we exit this function we
+        // can no longer modify the response headers.
+        self.set_http_response_header("content-length", None);
+        Action::Continue
+    }
+
+    fn on_http_response_body(&mut self, original_size: usize, end_of_stream: bool) -> Action {
+        if !end_of_stream {
+            // This returns "StopIterationAndBuffer" to Envoy, which will buffer the body.
```

Comment:
The Proxy-Wasm effort is proxy-agnostic, so I'd prefer if you didn't refer to Envoy internals here.

You could change it to something like:
```
// Wait until the complete request body is buffered on the host side.
```